src/Data/Lens/Lens.purs

```diff
@@ -37,6 +35,10 @@ withLens l f = case l (Shop id \_ b -> b) of Shop x y -> f x y
 cloneLens :: forall s t a b. ALens s t a b -> Lens s t a b
 cloneLens l = withLens l \x y p -> lens x y p
 
+both :: forall s a b c d. Lens s s a b -> Lens s s c d -> Lens s s (Tuple a c) (Tuple b d)
```

Comment:
This is not a valid lens in general if the two targets overlap.

Reply:
Hmm, you're right, I don't think it can be fixed though.  I added it last minute after realising that `takeBoth` could be generalised. I'll remove it from the PR.